recipes/libgpiod/all/conanfile.py

```diff
@@ -0,0 +1,75 @@
+from conans import ConanFile, tools, AutoToolsBuildEnvironment
+from conans.errors import ConanInvalidConfiguration
+import os
+import platform
+import re
+
+required_conan_version = ">=1.33.0"
+
+class LibgpiodConan(ConanFile):
+    name = "libgpiod"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://git.kernel.org/pub/scm/libs/libgpiod/libgpiod.git/"
+    license = "LGPL-2.1-or-later"
+    description = "C library and tools for interacting with the linux GPIO character device"
+    topics = ("conan", "gpio", "libgpiod", "libgpiodcxx", "linux")
+    settings = "os", "arch", "compiler", "build_type"
+    options = {"shared": [True, False],
+               "enable_bindings_cxx": [True, False],
+               "enable_tools": [True, False]}
+    default_options = {"shared": False,
+                       "enable_bindings_cxx": False,
+                       "enable_tools": False}
+    
+    _autotools = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def validate(self):
+        if self.settings.os != "Linux":
+            raise ConanInvalidConfiguration("only Linux is supported")
+        linux_kernel_version = re.match("([0-9.]+)", platform.release()).group(1)
+        if tools.Version(linux_kernel_version) < "4.8":
+            raise ConanInvalidConfiguration("only Linux kernel versions >= 4.8 are supported")
```

Comment:
It's always fragile if a validation check does not depend on a profile property.

Does conan have functionality for this? @jgsogo

Reply:
In the mean time, please add a `FIXME`.
e.g.
```python
# FIXME: this checks a non-profile property
```

Reply:
Does it require Linux >= 4.8 for building, running or both?

Reply:
In general, we cannot assume native building and, of course, we cannot assume that `conan info` command will be executed in the machine where it is going to be built (or run)... so we shouldn't use anything from the running machine as it will (in the general case) lead to wrong results.

We should always use the information provided by the user, which is `settings` and `options`. Here we should probably use some `os.version` or `os.kernel` setting the same way we are using `compiler.cppstd`: if it is specified by the user we validate it, if not we just assume it will work (the build-system will add required flags).

If we decide to use this _new_ setting, I think we need to add it first to the default Conan `settings.yml` file so everyone uses the same name and structure (although we won't be using it in CCI because it will compute different packageIDs).

Reply:
So do I just drop the check because there is no standardized profile variable for the kernel version?

Reply:
I would add more information to the error message, following the question by @Croydon: is it required to build or to run? But whatever you choose can be fine for this first PR, if we get issues about it, we can improve the recipe later.

---

Anyway, I think the underlying discussion is interesting (in general, not just for this PR) and maybe it's useful as a setting. I would like to hear others' opinions.

Reply:
The problem is I always get `configure: error: linux/gpio.h header not found (needed to build the library)` in the CI

Reply:
Please add the FIXME.
This is a problem that is out-of-scope for this recipe/pr specifically.

Reply:
there is a misunderstanding, it's not about the linux kernel version in container or host.
it's about providing linux kernel sources (headers) to the library.
see also: https://github.com/brgl/libgpiod/blob/master/README#L35
```
Recent kernel headers are also required for the GPIO user API definitions. For
the exact version of kernel headers required, please refer to the configure.ac
contents.
```
they need to be installed somewhere in the system, and we (obviously) don't have them installed in our CI right now.
it's not the first case, e.g. `v4l` (video for linux) also needs kernel headers for build.
one more time, it's not about checking the kernel version build machine is running, so the check for `platform.release()` doesn't solve anything and not needed.
for instance, you should be able to cross-compile this project from Windows or other system by just supplying correct linux kernel source (headers) downloaded.

so, what should be done and how?
1. this recipe should be `blocked` for a while
2. the condition `if tools.Version(linux_kernel_version) < "4.8":` should be removed altogether
3. new package `linux-kernel-headers/4.8.17` should be created first (it should be a header only package packaging headers from release tarball https://cdn.kernel.org/pub/linux/kernel/v4.x/linux-4.8.17.tar.gz)
4. recipe should add a reference `self.requires("linux-kernel-headers/4.8.17")`
5. recipe should pass include directories of `linux-kernel-headers/4.8.17` to the `./configure` script

Reply:
@SSE4 thanks for the insights. So the difference between this and `liburing` is that `liburing` actually uses new system calls introduced in kernel version 5.1 but `libgpiod` does not need any new system calls?

Reply:
Here's my attempt https://github.com/conan-io/conan-center-index/pull/6821 @SSE4